python/katana/example_utils.py

```diff
@@ -1,26 +1,34 @@
 import os
+from pathlib import Path
 import tarfile
 import urllib.request
+import shutil
 
 __all__ = ["get_input"]
 
 
-def get_cache_directory():
-    # Use the CI paths if they exist, just to make CI integration easier.
-    ci_inputs_path = os.environ["HOME"] + "/.cache/graph"
-    if os.path.isdir(ci_inputs_path):
-        return ci_inputs_path
-    # Otherwise use a more specific path name
-    return os.environ["HOME"] + "/.cache/galois"
-
-
-def get_inputs_directory():
-    cache_dir = get_cache_directory()
-    inputs_dir = cache_dir + "/inputs"
-    if os.path.isdir(inputs_dir) and os.path.isdir(inputs_dir + "/propertygraphs/ldbc_003"):
+def get_inputs_directory(*, invalidate=False):
+    inputs_dir = None
+    # Use the build paths if they exist.
+    paths_to_check = list(Path(__file__).parents) + list(Path.cwd().parents)
+    for path in paths_to_check:
+        # TODO(amp): If we can abstract the input version info a shared file, this should look for
+        #  specifically that version.
+        ci_inputs_path = path / "inputs" / "current"
+        if ci_inputs_path.is_dir():
+            inputs_dir = ci_inputs_path
+    # Otherwise use a cache directory
+    if not inputs_dir:
+        inputs_dir = Path(os.environ["HOME"]) / ".cache" / "katana" / "inputs"
+    if not invalidate and inputs_dir.is_dir() and (inputs_dir / "propertygraphs" / "ldbc_003").is_dir():
         return inputs_dir
+    try:
+        shutil.rmtree(inputs_dir)
+    except OSError:
+        inputs_dir.unlink()
+    inputs_dir.mkdir(parents=True, exist_ok=True)
```

Comment:
```python
if inputs_dir.is_dir() and (inputs_dir / "propertygraphs" / "ldbc_003").is_dir():
  if not invalidate:
    return inputs_dir
  try:
    shutil.rmtree(inputs_dir)
  except OSError:
    inputs_dir.unlink()
inputs_dir.mkdir(parents=True, exist_ok=True)
```
My best guess here is that this is the order of operations you want, since the "does not exist" case seemed to be raising `FileNotFoundError` on both `rmtree` and `unlink`. If that's wrong, then I think the reversion you describe is the best option

Reply:
I think you may be right. You can make that change and push to my branch. You have access. Or just do the roll back. Up to you.